src/domain/hanke/edit/utils.ts

```diff
@@ -166,3 +174,31 @@ export function getAreaDefaultName(areas?: HankeAlueFormState[]) {
 
   return `Hankealue ${maxAreaNumber + 1}`;
 }
+
+/**
+ * Sorts HAITTOJENHALLINTATYYPPI based on given tormaystarkasteluTulos.
+ * For example, if tormaystarkasteluTulos has {autoliikenneindeksi: 1.0, pyoraliiikenneindeksi: 1.3, linjaautoliikenneindeksi: 0.0, raitioliikenneindeksi: 0.0},
+ * the result will be [PYORALIIKENNE, AUTOLIIKENNE, LINJAAUTOLIIKENNE, RAITIOLIIKENNE].
+ */
+export function sortedLiikenneHaittojenhallintatyyppi(
+  tormaystarkasteluTulos: HankeIndexData,
+): HAITTOJENHALLINTATYYPPI[] {
+  const sortedIndices = Object.values(HANKE_INDEX_TYPE)
```

Comment:
Throws an error and app crashes if tormaystarkasteluTulos is undefined

Reply:
Refactored to handle `undefined` too and also to use enum order instead of alphabetical in case of equal number values. Added a test for `undefined`.